tempesta_fw/t/functional/helpers/analyzer.py

```diff
@@ -0,0 +1,123 @@
+"""
+Instruments for network traffic analysis
+"""
+from __future__ import print_function
+import os
+from threading import Thread
+from scapy.all import *
+from . import remote, tf_cfg
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+
+FIN = 0x01
+SYN = 0x02
+RST = 0x04
+PSH = 0x08
+ACK = 0x10
+URG = 0x20
+ECE = 0x40
+CWR = 0x80
+
+
+class Sniffer(object):
+
+    def __init__(self, node, host, count=0,
+                 timeout=30, port=80,
+                 node_close=True):
+        self.node = node
+        self.port = port
+        self.thread = None
+        self.captured = 0
+        self.packets = []
+        self.dump_file = '/tmp/tmp_packet_dump'
+        cmd = 'timeout %s tcpdump -i any %s-w - tcp port %s || true'
+        count_flag = ('-c %s ' % count) if count else ''
+        self.cmd = cmd % (timeout, count_flag, port)
+        self.err_msg = ' '.join(["Can't %s sniffer on", host])
+        self.node_side_close = node_close
+
+    def sniff(self):
+        stdout, stderr = self.node.run_cmd(self.cmd, timeout=None,
```

Comment:
Uh. I really dislike this tossing of data around: reading it completely in memory, then writing to a temporary file and finally reading it in memory again via scapy. How big is this dump expected to be on average, by the way?

It would be a lot nicer to feed the stream directly to scapy. https://github.com/secdev/scapy/issues/196 suggests you can pass an opened file object to `sniff(offline=...)` and it will parse the stream. Could you please try if scapy is happy with a file-like object returned by `paramiko.client.SSHClient.exec_command()`? 

This will probably require adding a new interface to `helpers.remote.Node` (it should relate to `helpers.remote.Node.exec_command()` just like `subprocess.Popen()` relates to `subprocess.run()`, i. e. run a command asynchronously and return its streams).

Reply:
Even if that fails somewhy, at least use StringIO to avoid having a temporary file.

Reply:
Unfortunately, `scapy.sniff(offline=file_obj)` interface does not support `StringIO` objects or `paramiko` file objects. I mentioned this in comment to `sniff()` function.